src/item_group.cpp

```diff
@@ -288,19 +289,21 @@ bool Single_item_creator::remove_item( const itype_id &itemid )
     return type == S_NONE;
 }
 
-void Single_item_creator::replace_item( const itype_id &itemid, const itype_id &replacementid )
+void Single_item_creator::replace_items( const std::unordered_map<itype_id, itype_id>
+        &replacements )
 {
     if( modifier ) {
-        modifier->replace_item( itemid, replacementid );
+        modifier->replace_items( replacements );
     }
     if( type == S_ITEM ) {
-        if( itemid.str() == id ) {
-            id = replacementid.str();
+        auto it = replacements.find( itype_id( id ) );
+        if( it != replacements.cend() ) {
```

Comment:
```suggestion
        if( it != replacements.end() ) {
```
Using `cend` is equivalent to `end`, and less idiomatic.